### PR TITLE
EE 225 approvers to jira

### PR DESCRIPTION
**Pull request by duanebester**

### CODE REVIEW

File: action.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      this.core.debug(
+        `Adding Jira Users as Code Reviewers: ${JSON.stringify(
+          resp.values.map((user) => user.displayName)
+        )}`
+      );
+      this.Jira.addCodeReviewersToIssue(this.issue.key, resp.values);
+    }
+  }
+
+  async updateApprovers() {
```
Comment (himichaelroberts) [inline action.js]: Where does this get called at?

File: index.js [hunk 2]
```diff
@@ -1,80 +1,51 @@
...
     }
 
-    const reviewers = action.getCodeReviewers();
-    const rigupReviewers = await Promise.all(
-      reviewers.map(async reviewer => {
-        return dynamo.findByGithubId(reviewer.id);
-      })
-    );
+    action.updateCodeReviewers();
+    action.updateApprovers();
```
Comment (duanebester) [inline index.js]: @himichaelroberts

Comment (duanebester) [reply index.js]: Don't worry about failed Checks, this will resolve those

File: lib/jira.js [hunk 3]
```diff
@@ -47,10 +47,10 @@ module.exports = class {
   }
 
   async getUsersFromAccountIds(accountIds) {
-    const query = { accountId: accountIds };
+    const query = { maxResults: 25, accountId: accountIds };
```
Comment (himichaelroberts) [inline lib/jira.js]: Why is maxResults needed?

Comment (duanebester) [reply lib/jira.js]: @himichaelroberts it defaulted to 10, so I wanted to have a safer, higher number

File: action.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    const commits = await this.getCommits();
+    const masterMergeStart = [
+      "Merge branch 'master'",
+      `Merged master into ${this.githubEvent.pull_request.head.ref}`,
+    ];
+    const originMergeStart = [
+      "Merge remote-tracking branch 'origin/master'",
+      `Merge remote-tracking branch '${this.githubEvent.pull_request.head.ref}'`,
+    ];
+    const conflictResolutionStart = [
```
Comment (reaganthomas) [inline action.js]: eyyy

File: action.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    const { data } = await this.octkit.pulls.listCommits({
+      owner: GITHUB_OWNER,
+      repo: this.githubEvent.repository.name,
+      pull_number: this.githubEvent.number,
+    });
+    return data;
+  }
+
+  async getIssues() {
```
Comment (reaganthomas) [inline action.js]: somewhat confusing the a method called `getIssues` returns an object containing 1 issue

File: action.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    // eslint-disable-next-line no-restricted-syntax
+    for (const issueKey of this.issueIds) {
+      // eslint-disable-next-line no-await-in-loop
+      const issue = await this.Jira.getIssue(issueKey);
+      if (issue) {
+        this.issue = issue;
```
Comment (reaganthomas) [inline action.js]: Unexpected side effect here

File: action.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    }
+    return {};
+  }
+
+  async updateCodeReviewers() {
+    const reviewers = this.getCodeReviewers();
+    const rigupReviewers = await Promise.all(
+      reviewers.map(async (reviewer) => {
+        return this.dynamo.findByGithubId(reviewer.id);
+      })
```
Comment (reaganthomas) [inline action.js]: Do we have linting enabled? Should see a semicolon here

File: action.js (hunk 6 above)
Comment (duanebester) [reply action.js]: Yeah-- will refactor this method

File: action.js (hunk 7 above)
Comment (duanebester) [reply action.js]: 🤔 we have linting... hmm

Comment (duanebester) [reply action.js]: @reaganthomas it's b/c it's inside the `Promise.all()` 😂

**Pull request merged**